packages/expo-notifications/src/scheduleNotificationAsync.ts

```diff
@@ -191,3 +186,86 @@ function isSecondsPropertyMisusedInCalendarTriggerInput(
       Object.keys(triggerWithoutChannelId).length > 1)
   );
 }
+
+function validateDateComponents(
+  object: any,
+  components: {
+    day?: boolean;
+    month?: boolean;
+    weekday?: boolean;
+    hour?: boolean;
+    minute?: boolean;
+  }
```

Comment:
is there a use case where we would need to provide `"<component>": false`?

might be cleaner to make this accept an array of strings, then we already have the `expectedProperties` without having to loop through them and filter for `true` or `false`

Reply:
No, there is no use case for false at the moment… I actually had this with an array initially… and yes, you can save the `expectedProperties` creation, but then you need to wrap the conditions below for each specific validation in a loop/forEach, and unless we use a set, there is a possibility (granted, bad usage, but still) of repeated keys and hence duplicate validation. So the typed object/dict/hash here ensured me that it wasn't possible to specify which components to validate in a wrong way.

What do you think? I don't mind changing it to any option that you think it suits the rest of the framework best.

Reply:
* forgot to mention that although it's not in use at the moment, in object form allows you to build the components to evaluate from conditions, like `{ day: isYearlyTrigger(trigger), month: isYearlyTrigger(trigger) }` for example, similar to one of the options from the [clsx](https://www.npmjs.com/package/clsx) package. But as mentioned, not an strong opinion for neither of us, so just let me know what you think it's best for the project

Reply:
> you need to wrap the conditions below for each specific validation in a loop/forEach

I think it's just one loop to iterate through the array:
```
For each component
 switch(component)
  case 'day':
  Validate day
  case 'month':
  Validate month
  // etc.
```

Reply:
yeah, I meant that single loop… still, with the mess of typings (using `NotificationTriggerInput` as the type of the function's argument) I had to make more changes here, so I'm turning it into a single loop and checking inside of the loop for both, undefined and correct range… I hope it turns out better